sdk/iothub/src/service/requests/get_twin_builder.rs

```diff
@@ -0,0 +1,62 @@
+use std::convert::TryFrom;
+use std::convert::TryInto;
+use std::marker::PhantomData;
+
+use bytes::Bytes;
+use http::{Method, Response, StatusCode};
+
+use crate::service::{IoTHubError, ServiceClient, API_VERSION};
+
+/// The GetTwinBuilder is used to construct a request to get the twin
+/// of a module or device.
+pub struct GetTwinBuilder<'a, T>
+where
+    T: TryFrom<Response<Bytes>, Error = IoTHubError>,
+{
+    service_client: &'a ServiceClient,
+    device_id: String,
+    module_id: Option<String>,
+    return_type: PhantomData<T>,
+}
+
+impl<'a, T> GetTwinBuilder<'a, T>
+where
+    T: TryFrom<Response<Bytes>, Error = IoTHubError>,
```

Comment:
Can you talk about why you did it this way (i.e, a generic on the builder itself  together with `PhantomData`) instead of making the `execute` method generic?

Reply:
Sure, by using `PhantomData` I was able to hide what the return type should be for the user. If I were to make this generic in the `execute` method, any user that calls any `getter` would have to specify the return type. 
```rust
// Original case:
let twin = service.get_device_twin("some-device").execute()

// Making execute generic:
let twin: DeviceTwinResponse = service.get_device_twin("some-device").execute();

// then this would also compile but fail horribly:
let twin: ModuleTwinResponse = service.get_device_twin("some-device").execute();
```
It would be much more error prone in my opinion (unless I'm missing a way this could be implemented by making `execute` generic ofcourse). Anyway, since I removed `execute()` from the getters this sort of changed by now. :)